packages/contracts/contracts/discovery/TestProjectVoting.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.12;
+
+import {ProjectVoting} from "./ProjectVoting.sol";
+
+import "hardhat/console.sol";
+
+contract TestProjectVoting is ProjectVoting {
+    Period public votingPeriod;
+    address[] public projects;
+    uint256 public numSlots;
+
+    constructor(
+        Period memory _votingPeriod,
+        address[] memory _projects,
+        uint256 _numSlots
+    ) ProjectVoting(_projects) {
+        votingPeriod = _votingPeriod;
+        projects = _projects;
+        numSlots = _numSlots;
+    }
+
+    function projectVoting_projects()
+        public
+        view
+        override(ProjectVoting)
+        returns (address[] memory)
+    {
+        return projects;
+    }
+
+    function projectVoting_votingPeriod()
+        public
+        view
+        override(ProjectVoting)
+        returns (Period memory)
+    {
+        return votingPeriod;
+    }
+
+    function projectVoting_singleSlotDuration()
+        public
+        view
+        override(ProjectVoting)
+        returns (uint256)
+    {
+        return (votingPeriod.end - votingPeriod.start) / numSlots;
+    }
+
+    function projectVoting_initialBonus()
+        public
+        view
+        override(ProjectVoting)
+        returns (int256)
+    {
+        return 0.05 * 10**18;
+    }
+
+    function projectVoting_finalBonus()
+        public
+        view
+        override(ProjectVoting)
+        returns (int256)
+    {
+        return 0;
+    }
+
+    function vote(address projectAddress) external {
+        _vote(projectAddress);
+    }
+
+    function getWinners() external view returns (address[] memory) {
+        return _getWinners();
+    }
+
+    function calculateWeightedVote(uint256 currentTime)
+        external
+        view
+        returns (uint256)
+    {
+        return _calculateWeightedVote(currentTime);
+    }
```

Comment:
can we prefix these 3 functions with `test_`? I find it helps distinguish real public functions from ones created by this test contract